daemon/create.go

```diff
@@ -139,6 +139,15 @@ func (daemon *Daemon) VolumeCreate(name, driverName string, opts map[string]stri
 		name = stringid.GenerateNonCryptoID()
 	}
 
+	daemon.volumes.mu.Lock()
```

Comment:
@cpuguy83 I guess the problem with that is that you'd still end up creating the volume, and we want to avoid it.
I agree though that this might not be the right place to put this code. What do you think about adding this inside `volume.Create` and returning an "already created" error?